src/primitive.rs

```diff
@@ -591,6 +591,62 @@ pub const fn any<'a, I: Input<'a>, E: ParserExtra<'a, I>>() -> Any<I, E> {
     }
 }
 
+/// See [`any_ref`].
+pub struct AnyRef<I, E> {
+    #[allow(dead_code)]
+    phantom: EmptyPhantom<(E, I)>,
+}
+
+impl<I, E> Copy for AnyRef<I, E> {}
+impl<I, E> Clone for AnyRef<I, E> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+impl<'a, I, E> ParserSealed<'a, I, &'a I::Token, E> for AnyRef<I, E>
+where
+    I: BorrowInput<'a>,
+    E: ParserExtra<'a, I>,
+{
+    #[inline]
+    fn go<M: Mode>(&self, inp: &mut InputRef<'a, '_, I, E>) -> PResult<M, &'a I::Token> {
+        let before = inp.offset();
+        match inp.next_ref_inner() {
+            (_, Some(tok)) => Ok(M::bind(|| tok)),
+            (at, found) => {
+                let err_span = inp.span_since(before);
+                inp.add_alt(at, None, found.map(|f| f.into()), err_span);
+                Err(())
+            }
+        }
+    }
+
+    go_extra!(&'a I::Token);
+}
+
+/// A parser that accepts any input (but not the end of input).
+///
+/// The output type of this parser is `I`, the input that was found.
```

Comment:
I think this should read `&'a I::Token` instead of `I`.

It might also be worth adding something like this:

> This function is the borrowing equivalent of [`any`]. Where possible, it's recommended to use [`any`] instead.

Reply:
Good comment thanks, and fixed!